src/lndk_offers.rs

```diff
@@ -247,8 +228,12 @@ impl OfferHandler {
                 }
             };
         }
-
-        Ok((invoice_request, payment_id, validated_amount))
+        let offer_context = OffersContext::OutboundPayment {
+            payment_id,
+            nonce,
+            hmac: None,
```

Comment:
You'll want to include an HMAC of the `payment_id` here. That way, if the recipient sends back an `InvoiceError` you can authenticate the `payment_id` before using it to log or clean up state.